lobby/src/main/java/org/triplea/lobby/server/login/forgot/password/create/TempPasswordPersistence.java

```diff
@@ -0,0 +1,40 @@
+package org.triplea.lobby.server.login.forgot.password.create;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.function.Function;
+import javax.annotation.Nonnull;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import org.mindrot.jbcrypt.BCrypt;
+import org.triplea.lobby.common.login.RsaAuthenticator;
+import org.triplea.lobby.server.db.HashedPassword;
+import org.triplea.lobby.server.db.JdbiDatabase;
+import org.triplea.lobby.server.db.dao.TempPasswordDao;
+
+/**
+ * Stores a user temporary password in database. When we generate a new temporary password, all
+ * existing temporary passwords are invalidated so that a user only has one temp password at a time.
+ */
+@AllArgsConstructor(
+    access = AccessLevel.PACKAGE,
+    onConstructor_ = {@VisibleForTesting})
+class TempPasswordPersistence {
+  @Nonnull private final TempPasswordDao tempPasswordDao;
+  @Nonnull private final Function<String, HashedPassword> passwordHasher;
+  @Nonnull private final Function<HashedPassword, String> hashedPasswordBcrypter;
+
+  static TempPasswordPersistence newInstance() {
+    return new TempPasswordPersistence(
+        JdbiDatabase.newConnection().onDemand(TempPasswordDao.class),
+        pass -> new HashedPassword(RsaAuthenticator.hashPasswordWithSalt(pass)),
+        hashedPass -> BCrypt.hashpw(hashedPass.value, BCrypt.gensalt()));
+  }
+
+  boolean storeTempPassword(final String username, final String generatedPassword) {
+    tempPasswordDao.invalidateTempPasswords(username);
+
+    final HashedPassword hashedPass = passwordHasher.apply(generatedPassword);
+    final String tempPass = hashedPasswordBcrypter.apply(hashedPass);
+    return tempPasswordDao.insertTempPassword(username, tempPass);
```

Comment:
To be fair I don't think it's necessary to hash a random token that doesn't originate from user input and expires after a short amount of time, but if you want to treat it like a password to keep all the other logic the same that's fine too I guess.

Reply:
There actually was no expiration. I added in one here: f23c3e5ae694b1e5f8ffe457ae67ad831603a5a8

We would want to treat the temp token with the same security. If someone can read DB values, and we had a weaker hashing strategy, it'd be an attack vector. Instead of cracking the password hashes, someone would just need to request a password reset and then they could crack the easier hash instead.